src/catchup/simulation/ApplyTransactionsWork.cpp

```diff
@@ -63,14 +64,125 @@ checkResults(Application& app, uint32_t ledger,
         auto const& res = results[i];
         assert(res.transactionHash == resSet.results[i].transactionHash);
 
-        auto const& dbRes = resSet.results[i].result;
-        if (dbRes.result.code() != res.result.result.code())
+        auto const& dbRes = resSet.results[i].result.result;
+        if (dbRes.code() != res.result.result.code())
         {
             auto msg = fmt::format(
                 "Expected result code {} does not agree with {} for tx {}",
-                res.result.result.code(), dbRes.result.code(),
+                res.result.result.code(), dbRes.code(),
                 binToHex(res.transactionHash));
             CLOG(ERROR, "History") << msg;
+            return;
+        }
+
+        // Now check each operation result
+        auto const& archiveRes = results[i].result.result;
+
+        if (dbRes.code() == txFAILED || dbRes.code() == txSUCCESS)
+        {
+            assert(archiveRes.code() == txFAILED ||
+                   archiveRes.code() == txSUCCESS);
+            assert(archiveRes.results().size() == dbRes.results().size());
+
+            for (size_t j = 0; j < archiveRes.results().size(); j++)
+            {
+                if (archiveRes.results()[j].code() != dbRes.results()[j].code())
+                {
+                    throw std::runtime_error(
+                        fmt::format("Expected {} but got {}",
+                                    archiveRes.results()[j].code(),
```

Comment:
This is going to log numbers like "Expected -3 but got -5". Is that what you intended? It's possible to get the actual strings if you want them.